memtable/write_buffer_manager.cc

```diff
@@ -217,20 +219,37 @@ void WriteBufferManager::MaybeFlushLocked(DB* this_db) {
         current_score = current_score * (100 - factor) / factor;
       }
     }
-    if (current_score > max_score) {
-      candidate = s.get();
-      max_score = current_score;
-      candidate_size = current_memory_bytes;
+    for (size_t i = 0; i < kCandidateSize; i++) {
+      if (current_score > std::get<1>(candidates[i])) {
+        candidates[i] = {s.get(), current_score, oldest_time};
+        break;
```

Comment:
It might miss the second large candidate. Eg., for scores 2, 1, 3, it may end with [3, 1].